demos/random-values-stream/index.js

```diff
@@ -0,0 +1,217 @@
+/**
+ * Creates random values with the help of WebCrypto API
+ */
+function createRandomValuesStream(numberOfBytes = 10, valueInterval = 1000, maxValues = null) {
+    const cqs = new CountQueuingStrategy({ highWaterMark: 4 });
+    const readableStream = new ReadableStream({
+        totalEnqueuedItemsCount: 0,
+        interval: null,
+
+        start(controller) {
+            logStatusText('`start` method of the readable stream called')
+            this.startValueInterval(controller);
+        },
+
+        /**
+         * Starting the random values generation again after a certain period
+         * @param {*} controller 
+         */
+        async pull(controller) {
```

Comment:
This is unnecessarily complicated. You don't need to track `desiredSize` yourself. As long as you have called `enqueue()` you can rely on the ReadableStream to call you back when there is no backpressure.

It would be simpler to do something like

```javascript
pull(controller) {
  if (maxValues && this.totalEnqueuedItemsCount >= maxValues) {
    controller.close();
    return;
  }
  let resolve;
  const promise = new Promise(r => {
    resolve = r;
  });
  setTimeout(() => {
    controller.enqueue(randomValuesUint8Array(20));
    ++this.totalEnqueuedItemsCount;
    resolve();
  }, valueInterval);
  return promise;
}
```

(untested).